mat/svd_test.go

```diff
@@ -212,3 +212,451 @@ func extractSVD(svd *SVD) (s []float64, u, v *Dense) {
+			}, wm: 3, wn: 1,
+		},
+		{
+			a: []float64{
+				0.8147, 0.9134, 0.231, -1.65,
+				0.9058, 0.6324, 0.9, 0.72,
+				0.1270, 0.0975, 0.1, 1.723,
+				1.6, 2.8, -3.5, 0.987,
+				7.231, 9.154, 1.823, 0.9,
+			}, m: 5, n: 4,
+			b: []float64{
+				0.278, 8.635,
+				0.547, 9.125,
+				-0.958, -0.762,
+				1.452, 1.444,
+				1.999, -7.234,
+			}, bc: 2,
+			want: []float64{
+				1.863006789511373, 44.467887791812750,
+				-1.127270935407224, -34.073794226035126,
+				-0.527926457947330, -8.032133759788573,
+				-0.248621916204897, -2.366366415805275,
+			}, wm: 4, wn: 2,
+		},
+		{
+			// Test rank-deficient case compared with numpy.
+			// >>> import numpy as np
+			// >>> b = np.array([[-2.3181340317357653],
+			// ...     [-0.7146777651358073],
+			// ...     [1.8361340927945298],
+			// ...     [-0.35699930593018775],
+			// ...     [-1.6359508076249094]])
+			// >>> A = np.array([[-1.7854591879711257, -0.42687285925779594, -0.12730256811265162],
+			// ...     [-0.5728984211439724, -0.10093393134001777, -0.1181901192353067],
+			// ...     [1.2484316018707418, 0.5646683943038734, -0.48229492403243485],
+			// ...     [0.10174927665169475, -0.5805410929482445, 1.3054473231942054],
+			// ...     [-1.134174808195733, -0.4732430202414438, 0.3528489486370508]])
+			// >>> np.linalg.lstsq(A, b, rcond=None)
+			// (array([[ 1.21208422],
+			//        [ 0.41541503],
+			//        [-0.18320349]]), array([], dtype=float64), 2, array([2.68451480e+00, 1.52593185e+00, 6.82840229e-17]))
+
+			a: []float64{
+				-1.7854591879711257, -0.42687285925779594, -0.12730256811265162,
+				-0.5728984211439724, -0.10093393134001777, -0.1181901192353067,
+				1.2484316018707418, 0.5646683943038734, -0.48229492403243485,
+				0.10174927665169475, -0.5805410929482445, 1.3054473231942054,
+				-1.134174808195733, -0.4732430202414438, 0.3528489486370508,
+			}, m: 5, n: 3,
+			b: []float64{
+				-2.3181340317357653,
+				-0.7146777651358073,
+				1.8361340927945298,
+				-0.35699930593018775,
+				-1.6359508076249094,
+			}, bc: 1,
+			cond: 1e-15,
+			want: []float64{
+				1.2120842180372118,
+				0.4154150318658529,
+				-0.1832034870198265,
+			}, wm: 3, wn: 1,
+		},
+		{
+			a: []float64{
+				0, 0,
+				0, 0,
+			}, m: 2, n: 2,
+			b: []float64{
+				3,
+				2,
+			}, bc: 1,
+		},
+		{
+			a: []float64{
+				0, 0,
+				0, 0,
+				0, 0,
+			}, m: 3, n: 2,
+			b: []float64{
+				3,
+				2,
+				1,
+			}, bc: 1,
+		},
+		{
+			a: []float64{
+				0, 0, 0,
+				0, 0, 0,
+			}, m: 2, n: 3,
+			b: []float64{
+				3,
+				2,
+			}, bc: 1,
+		},
+	} {
+		a := NewDense(test.m, test.n, test.a)
+		b := NewDense(test.m, test.bc, test.b)
+
+		var want *Dense
+		if test.want != nil {
+			want = NewDense(test.wm, test.wn, test.want)
+		}
+
+		var svd SVD
+		ok := svd.Factorize(a, SVDFull)
+		if !ok {
+			t.Errorf("unexpected factorization failure for test %d", i)
+			continue
+		}
+
+		var x Dense
+		rank := svd.Rank(test.cond)
+		if rank == 0 {
+			continue
```

Comment:
I don't know if this is triggered but add `t.Log`? If done, then also below.

Reply:
See https://github.com/gonum/gonum/pull/1316#discussion_r438056289 below.